bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraConnector.java

```diff
@@ -0,0 +1,223 @@
+			e.printStackTrace();
+		}
+		
+		defaultFolder = createFolder(projectName + "-Default", projectName + "-Default");
+		
+		this.logService.log(LogService.LOG_INFO, "Initialized Jira Connector with " + properties.toString());
+	}
+
+	private void validateConfig(Map<String, Object> properties) throws SpecmateValidationException {
+		String aProject = (String) properties.get(JiraConnectorConfig.KEY_JIRA_PROJECT);
+		String aUsername = (String) properties.get(JiraConnectorConfig.KEY_JIRA_USERNAME);
+		String aPassword = (String) properties.get(JiraConnectorConfig.KEY_JIRA_PASSWORD);
+
+		if (isEmpty(aProject) || isEmpty(aUsername) || isEmpty(aPassword)) {
+			throw new SpecmateValidationException("Jira Connector is not well configured.");
+		}
+	}
+
+	@Reference
+	public void setLogService(LogService logService) {
+		this.logService = logService;
+	}
+
+	@Override
+	public String getId() {
+		return this.id;
+	}
+
+	@Override
+	public Collection<Requirement> getRequirements() throws SpecmateException {
+		
+		List<Requirement> requirements = new ArrayList<>();
+		
+		List<Issue> storiesWithoutEpic = this.getStoriesWithoutEpic();
+		for (Issue story : storiesWithoutEpic) {
+			requirements.add(createRequirement(story, null));
+		}
+
+		List<Issue> epics = this.getEpics();
+		
+		for (Issue epic : epics) {
+			createFolderIfNotExists(epic);
+			List<Issue> stories = getStoriesForEpic(epic);
+			for (Issue story : stories) {
+				Requirement requirement = createRequirement(story, epic);
+				requirmentEpics.put(requirement, epic);
+				requirements.add(requirement);
+			}
+		}
+		
+		return requirements;
+	}
+	
+	private List<Issue> getStoriesForEpic(Issue epic) {
+		return this.getIssues("project=" + projectName + " AND issueType=story AND \"Epic Link\"=\"" + epic.getKey() + "\" ORDER BY assignee, resolutiondate");
+	}
+	
+	private List<Issue> getStoriesWithoutEpic() {
+		return this.getIssues("project=" + projectName + " AND issueType=story AND \"Epic Link\" IS EMPTY ORDER BY assignee, resolutiondate");
+	}
+	
+	private List<Issue> getEpics() {
+		return this.getIssues("project=" + projectName + " AND issueType=epic ORDER BY id");
+	}
+	
+	private List<Issue> getIssues(String jql) {
+		List<Issue> issues = new ArrayList<>();
+		
+		int maxResults = Integer.MAX_VALUE;
+		while(issues.size() < maxResults) {
+			try{
+				SearchResult searchResult = jiraClient.getSearchClient().searchJql(jql, -1, issues.size(), null).claim();
+				maxResults = searchResult.getTotal();
+				searchResult.getIssues().forEach(issue -> issues.add(issue));
+				logService.log(LogService.LOG_DEBUG, "Loaded ~" + searchResult.getMaxResults() + " issues from Jira " + url + " project: " + projectName);
+			} catch(RestClientException e) {
+				if(e.getStatusCode().get() == 400) {
+					// We expect this in the case, where no results are found.
+					logService.log(LogService.LOG_INFO, "Error while claiming results for query [" + jql + "]: " + e.getMessage());

```

Comment:
If this result is expected if no issues are found, then I would not use the term "Error" in the log message.